DependencyInjection/Compiler/MiddlewaresPass.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler;
+
+use Doctrine\Bundle\DoctrineBundle\Middleware\ConnectionNameAwareInterface;
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+use function array_keys;
+use function is_subclass_of;
+use function sprintf;
+
+class MiddlewaresPass implements CompilerPassInterface
+{
+    /** @var string  */
+    private $connexionDefsParam;
+
+    /** @var string  */
+    private $middlewareTag;
+
+    public function __construct(
+        string $connexionDefsParam = 'doctrine.connections',
+        string $middlewareTag = 'doctrine.middleware'
+    ) {
+        $this->connexionDefsParam = $connexionDefsParam;
+        $this->middlewareTag      = $middlewareTag;
+    }
+
+    public function process(ContainerBuilder $container)
+    {
+        $middlewareAbstractDefs = [];
+        foreach (array_keys($container->findTaggedServiceIds($this->middlewareTag)) as $id) {
+            $def                         = $container->getDefinition($id);
+            $middlewareAbstractDefs[$id] = $def;
+        }
+
+        $connections = $container->getParameter($this->connexionDefsParam);
+        foreach ($connections as $name => $id) {
+            $middlewareDefs = [];
+            foreach ($middlewareAbstractDefs as $id => $abstractDef) {
+                $middlewareDefs[] = $childDef = $container->setDefinition(
+                    sprintf('%s.%s', $id, $name),
+                    new ChildDefinition($id)
+                );
+
+                if (! is_subclass_of($abstractDef->getClass(), ConnectionNameAwareInterface::class)) {
+                    continue;
+                }
+
+                $childDef->addMethodCall('setConnectionName', [$name]);
```

Comment:
I get the need for early continue/return, etc. but here it seems needlessly convoluted as it requires a logical inversion and an extra statement.

```suggestion
                if (is_subclass_of($abstractDef->getClass(), ConnectionNameAwareInterface::class)) {
                    $childDef->addMethodCall('setConnectionName', [$name]);
                }
```